src/test/java/com/github/messenger4j/test/integration/DocumentationTest.java

```diff
@@ -203,4 +215,67 @@ public void echoExample() throws Exception {
         });
         // end::doc-EchoExample[]
     }
+    
+    @ToString(callSuper = true)
+    @EqualsAndHashCode(callSuper = true)
+    public static class CustomEvent extends BaseEvent {
+    	
+    	@Getter
+    	@Accessors(fluent=true)
+    	private String custom;
+
+		CustomEvent(String custom) {
+			super(null, null, null);
+			this.custom = custom;
+		}
+    	
+    }
+    
+    public static class CustomEventFactory implements BaseEventFactory<CustomEvent> {
+
+		@Override
+		public boolean isResponsible(JsonObject messagingEvent) {
+			return messagingEvent.has("custom");
+		}
+
```

Comment:
Max used `Jsonhelper.getPropertyAsString(messagingEvent, CUSTOM).orEleseThrow(IllegalArgumentException::new)` for extracting a key out of JSON 👍